server/src/main/java/org/elasticsearch/cluster/NodeUsageStatsForThreadPools.java

```diff
@@ -36,86 +38,60 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeMap(threadPoolUsageStatsMap, StreamOutput::writeWriteable);
     }
 
-    @Override
-    public int hashCode() {
-        return Objects.hash(nodeId, threadPoolUsageStatsMap);
-    }
+    /**
+     * One utilization sample
+     *
+     * @param instant The time we received the sample
+     * @param utilization The utilization value in the range [0.0, 1.0]
+     */
+    public record UtilizationSample(Instant instant, float utilization) implements Writeable {
 
-    @Override
-    public boolean equals(Object o) {
```

Comment:
There were equals and hashcode methods on these records, I'm not sure if they were once classes or we're doing something special here? I removed them in lieu of the ones you get for free with a record.